control/pid_longitudinal_controller/src/pid_longitudinal_controller.cpp

```diff
@@ -783,8 +818,11 @@ double PidLongitudinalController::calcFilteredAcc(
   // store ctrl cmd without slope filter
   storeAccelCmd(acc_max_filtered);
 
+  // if current state is STOPPED, do not apply slope compensation to avoid sudden acceleration
```

Comment:
I don't know why the slope compensation causes a sudden stop. The sudden acceleration should not depend on the slope compensation, but the stop acceleration parameter, I guess.